### DISCUSSION TITLE
Where are provided texts(paragraphs) supposed to be?

### DISCUSSION THREAD

**Discussion by JhohannesK** [JhohannesK/TipoHiZ#21] [Ideas]
Should the text that would be randomly provided to the user to type be on the client side or on the server

**Comment by darkjoka** [comment]
I'm thinking on the client but that can make the loading very slow if it becomes many.
On the server side, clicking refresh for new phrases quickly with a slower internet might make the experience laggy.
I'll suggest we store the text on the server and fetch them in chunks like 15 or something. If we get close to exhausting what we have on the client, we can request another chunk. We get reduced server requests and snappy feel on the client. What do you think
